src/core/nvector.cc

```diff
@@ -0,0 +1,451 @@
+  // equation 10 (normalized, n_EA_E and kaeast are perpendicular unit vectors)
+  Vector3D kanorth = crossProduct(n_EA_E, kaeast);
+
+  // equation 11
+  // Ren = [kanorth(normalized) kaeast(normalized) -n_EA_E];
+  // and a rotation from the E frame to the N frame (North, East, Down).
+  // P_AB_N = R_EN' * p_AB_E
+  double P_AB_N_x = kanorth*p_AB_E;
+  double P_AB_N_y = kaeast*p_AB_E;
+  // double P_AB_N_z = (-n_EA_E)*p_AB_E;
+
+  double azimuth = atan2(P_AB_N_y, P_AB_N_x);
+  return azimuth;
+}
+
+double NVector::azimuth(const NVector& n_EA_E, const NVector& n_EB_E, double height_a, double height_b)
+{
+  double azimuth = azimuth_radians(n_EA_E, n_EB_E, height_a, height_b);
+  double azimuth_degrees = azimuth * 180.0/M_PI;
+  return azimuth_degrees;
+}
+
+// returns values in the range [0.0,360)
+double NVector::azimuth_true_degrees(const NVector& n_EA_E, const NVector& n_EB_E, double height_a, double height_b)
+{
+  double result = azimuth(n_EA_E, n_EB_E, height_a, height_b);
+  if (result < 0.0) {
+    result += 360.0;
+  }
+  if (result >= 360.0) {
+    result = 0.0;
+  }
+  return result;
+}
+
+double NVector::azimuth(double latitude_a_degrees, double longitude_a_degrees,
+                        double latitude_b_degrees, double longitude_b_degrees,
+                        double height_a, double height_b)
+{
+  NVector n_EA_E(latitude_a_degrees, longitude_a_degrees);
+  NVector n_EB_E(latitude_b_degrees, longitude_b_degrees);
+  return azimuth(n_EA_E, n_EB_E, height_a, height_b);
+}
+
+// returns values in the range [0.0,360)
+double NVector::azimuth_true_degrees(double latitude_a_degrees, double longitude_a_degrees,
+                                     double latitude_b_degrees, double longitude_b_degrees,
+                                     double height_a, double height_b)
+{
+  double result = azimuth(latitude_a_degrees, longitude_a_degrees,
+                          latitude_b_degrees, longitude_b_degrees,
+                          height_a, height_b);
+  if (result < 0.0) {
+    result += 360.0;
+  }
+  if (result >= 360.0) {
+    result = 0.0;
+  }
+  return result;
+}
+
+#if 0
+// This interpolation is nonlinear!
+NVector NVector::linepart(const NVector& n_EA_E, const NVector& n_EB_E, double fraction)
+{
+  Vector3D n_ER_E = (n_EA_E + (n_EB_E - n_EA_E)*fraction).normalize();
+  return n_ER_E;
+}
+#elif 0
+NVector NVector::linepart(const NVector& n_EA_E, const NVector& n_EB_E, double fraction)
+{
+  // see example 8.
+  double sab_rad = distance_radians(n_EA_E, n_EB_E);
+  double sar_rad = fraction*sab_rad;
+  double az_rad = azimuth_radians(n_EA_E, n_EB_E);
+
+  // equation 9 (normalized)
+  Vector3D kaeast = crossProduct(Vector3D(1.0, 0.0, 0.0), n_EA_E).normalize();
+  // equation 10 (normalized, n_EA_E and kaeast are perpendicular unit vectors)
+  Vector3D kanorth = crossProduct(n_EA_E, kaeast);
+
+  Vector3D d_EA_E = kanorth*cos(az_rad) + kaeast*sin(az_rad);
```

Comment:
Please run astyle or clang-tidy to format. This is one of many places that binary operand spacing is inconsistent. Here it's both wrong (*) and right (+) in the same line.

Reply:
The only thing astyle changes with our style file seems incorrect, it eliminates the space before the '*' in
WGS84_ECCENTRICITY_SQUARED = 1.0 - (WGS84_ASPECT_RATIO * WGS84_ASPECT_RATIO)

I don't want to get into formatting flip flops with two rulers, i.e. astyle and clang-format.

Reply:
astyle could do what I think you want with "--pad-oper", but that hasn't been part of our style file.